src/platforms/esp32/components/avm_sys/sys.c

```diff
@@ -546,13 +546,25 @@ term sys_get_info(Context *ctx, term key)
         term_set_map_assoc(ret, 3, globalcontext_make_atom(glb, revision_atom), term_from_int32(info.revision));
         return ret;
     }
-    if (key == globalcontext_make_atom(glb, esp_idf_version_atom)) {
-        const char *str = esp_get_idf_version();
-        size_t n = strlen(str);
-        if (memory_ensure_free(ctx, 2 * n) != MEMORY_GC_OK) {
+    if (globalcontext_is_term_equal_to_atom_string(glb, key, ATOM_STR("\x7", "os_type"))) {
+        if (UNLIKELY(memory_ensure_free(ctx, TUPLE_SIZE(2)) != MEMORY_GC_OK)) {
             return OUT_OF_MEMORY_ATOM;
         }
-        return term_from_string((const uint8_t *) str, n, &ctx->heap);
+        term os = term_alloc_tuple(2, &ctx->heap);
+        term_put_tuple_element(os, 0, ESP32_ATOM);
+        term_put_tuple_element(os, 1, globalcontext_make_atom(glb, ATOM_STR("\x7", "esp-idf")));
```

Comment:
I thought about this, we should return a {Platform, Flavor} tuple. E.g. {esp-idf, esp32}, but it might be {zephyr, esp32} or {zephyr, stm32}. @pguyot any opinion here?

Reply:
This would kind of break OTP output, which is just the OS. We have other means of determining the cpu architecture. I think if an esp32 port is added for zephyr that should just return Zephyr for the os.

Reply:
The os_type spec is:
```
-spec type() -> {Osfamily, Osname} when Osfamily :: unix | win32, Osname :: atom().
```

Reply:
I do follow your logic here, but I feel like we are getting very specific in the “general” names here, the general is extremely broad… it almost seems to me the most appropriate “general” to return for all of the MCUs would be “embedded” and then the “specific” names would be esp-idf, zephyr, pico-sdk, opencm3, etc…

I almost feel this is worth raising a question in the Erlang forums. There may be members of the Embedded Working Group that have already considered how OTP will handle this situation.

Reply:
The main reason I believe it is correct to return zephyr or esp-idf for the os name is that it is the version number for these that is returned for os_version just like the kernel version. So the os_version is paired to the “Specific” not the “Generic”… otherwise the return from the MCUs will differ from the return on the Unix port.